### ISSUE TITLE
Give permissions/roles based on multiple accounts

### ISSUE THREAD

**Issue by juliancc** [spatie/laravel-permission#1067]
Is there a way to give a permission/role to a user when using a multiple level account system.

Example:

Account 1
-- User 1 [Admin]
-- User 2 [Editor]

Account 2
-- User 1 [Editor]

The user is able to swap accounts during their session.

**Comment by drbyte** [comment]
Based on the hierarchy you posted, I assume that by "Account" you mean "Tenant"?

This package responds based on the relations loaded for the user object. So in your app you'd need to reload all the associations for the user after switching "Accounts".

**Comment by juliancc** [comment]
Not quiet multi-tenancy. Same user can belong to multiple accounts and have different roles/permissions. 
I think I solved it with some minor tweaks:

- First I added account_id to the model_has_roles model_has_permissions tables
- Then in the HasRoles HasPermissions traits the morph methods (roles, permissions) I included a where clause with the account_id
- Lastly, I created a Global Scope for my models that I attach as a trait that includes something like this: `$builder->where('account_id', $currentAccount->id);`

For now this works, was wondering if there was a cleaner way to do it. :D
